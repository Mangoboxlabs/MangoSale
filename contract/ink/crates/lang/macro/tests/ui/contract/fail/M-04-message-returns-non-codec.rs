use ink_lang as ink;

#[ink::contract]
mod message_returns_non_codec {
    #[derive(scale_info::TypeInfo)]
    pub struct NonCodec;

    #[ink(storage)]
    pub struct MessageReturnsNonCodecType {}

    impl MessageReturnsNonCodecType {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn returns_non_codec_type(&self) -> NonCodec {
            NonCodec
        }
    }
}

fn main() {}
