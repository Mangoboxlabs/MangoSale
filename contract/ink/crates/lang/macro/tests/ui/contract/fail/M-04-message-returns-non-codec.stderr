error[E0277]: the trait bound `NonCodec: WrapperTypeEncode` is not satisfied
  --> $DIR/M-04-message-returns-non-codec.rs:18:49
   |
18 |         pub fn returns_non_codec_type(&self) -> NonCodec {
   |                                                 ^^^^^^^^ the trait `WrapperTypeEncode` is not implemented for `NonCodec`
   |
   = note: required because of the requirements on the impl of `Encode` for `NonCodec`
note: required by a bound in `ink_lang::FnOutput::Output`
  --> $DIR/traits.rs:79:18
   |
79 |     type Output: scale::Encode + 'static;
   |                  ^^^^^^^^^^^^^ required by this bound in `ink_lang::FnOutput::Output`
